hooks/pnpmfile/src/requirePnpmfile.ts

```diff
@@ -62,18 +65,20 @@ export async function requirePnpmfile (pnpmFilePath: string, prefix: string): Pr
     }
     if (pnpmfile?.hooks?.readPackage) {
       const readPackage = pnpmfile.hooks.readPackage as Function // eslint-disable-line
-      pnpmfile.hooks.readPackage = async function (pkg: PackageManifest, ...args: any[]) { // eslint-disable-line
-        pkg.dependencies = pkg.dependencies ?? {}
-        pkg.devDependencies = pkg.devDependencies ?? {}
-        pkg.optionalDependencies = pkg.optionalDependencies ?? {}
-        pkg.peerDependencies = pkg.peerDependencies ?? {}
-        const newPkg = await readPackage(pkg, ...args)
+      pnpmfile.hooks.readPackage = async function <Pkg extends BaseManifest>(pkg: Pkg, ...args: any[]): Promise<Pkg> { // eslint-disable-line
```

Comment:
Some of the changes seem unrelated to the new resolvers/fetchers. Maybe create a separate PR for the unrelated refactor.